jdaviz/configs/imviz/plugins/tools.py

```diff
@@ -51,7 +51,10 @@ def on_click(self, data):
         if x is None or y is None:  # Out of bounds
             return
         x, y, _, _ = self.viewer._get_real_xy(image, x, y)
-        self.viewer.center_on((x, y))
+        if image.coords is not None:
+            self.viewer.center_on(image.coords.pixel_to_world(x, y))
```

Comment:
The use of `_get_real_xy` above ensures that the behavior for data with WCS (in the `else`) and without WCS (in the `if`) is consistent. `image` here is [the first visible layer](https://github.com/bmorris3/jdaviz/blob/131a4123e6e03fbd505dc9992a947c5f85510d0f/jdaviz/configs/imviz/plugins/tools.py#L48), which is not the reference data layer when 
linked by WCS. So this call tries to be explicit.

Reply:
If you want to switch the call to using sky coordinates, you don't need `_get_real_xy`. You can theoretically pass `(x, y)` directly to reference WCS because input `(x, y)` is always w.r.t. reference data.

Reply:
So do you want to see line 53 moved inside the conditional?

Reply:
If this line makes the final cut, I think you can change the call to something like `reference_data.coords.pixel_to_world(x, y)` without ever calling `viewer._get_real_xy(image, ...)`.

But I am not sure what you need to do to fix https://github.com/bmorris3/jdaviz/pull/29#discussion_r1450789355